Feature/18-logging (#20)
miraisolutions/secretsanta: README.md

* [Virtual environments](https://docs.python-guide.org/dev/virtualenvs/)
* [Creating virtual environments](https://packaging.python.org/tutorials/installing-packages/#creating-virtual-environments)

> Install support for virtual environments with Python 3.8.0 if you don't have it yet:

Reviewer: I think we need to re-phrase this comment in conjunction with the modified command below

> Install support for virtual environments with Python 3.8.0 if you don't have it yet:

Reviewer: ```suggestion
Install support for virtual environments with Python 3.x if you don't have it yet. Below we assume Python 3.8:
```

```bash

> sudo apt install python3.8

Reviewer: ```suggestion
sudo apt-get install python3.8-venv
```

### Miscellaneous

##### Logging

The `logging` package is used to track events after running the project. The main logged events (levels) in Secret Santa are: errors, warnings, and participants info. A log level is set as an environment variable, e.g.:
```bash
os.environ["level"] = "ERROR"
```

All logs activities are collected into a log file that is initiated at the beginning of the code:
```bash

> logging.basicConfig(filename = 'file_name')

Reviewer: maybe you would like to update this command (?)

Author: command inputs have been updated

```
A logger is then set:
```bash
logger = logging.getLogger(__name__)
```
All functions used afterwards refer to this logger:
```bash
logger.error("Error message")
logger.warning("Warning message")
logger.info("Info")
```

> The log file is automatically created in the current directory and can be inspected  after the project run is complete. 

Reviewer: is this ("current directory") still correct?

Author: sentence has been edited

### Miscellaneous

##### Logging

The `logging` package is used to track events after running the project. The main logged events (levels) in Secret Santa are: errors, warnings, and participants info. A log level is set as an environment variable, e.g.:
```bash
os.environ["level"] = "ERROR"
```

All logs activities are collected into a log file that is initiated at the beginning of the code:
```bash
logging.basicConfig(filename = path_to_file, level = level, format = '%(asctime)s %(levelname)s %(message)s',
                               datefmt = '%Y/%m/%d %I:%M:%S %p')
```
A logger is then set:
```bash
logger = logging.getLogger(__name__)
```
All functions used afterwards refer to this logger:
```bash
logger.error("Error message")
logger.warning("Warning message")
logger.info("Info")
```

> The log file is automatically created in the `log_files` directory and can be inspected  after the project run is complete. 

Reviewer: ```suggestion
The log file is automatically created in the `log_files` directory and can be inspected after the project run is complete. 
```
